force_bdss/mco/optimizer_engines/weighted_optimizer_engine.py

```diff
@@ -55,15 +56,23 @@ def sen_scaling_method(dimension, weighted_optimize):
     return scaling_factors
 
 
-class WeightedOptimizerEngine(ScipyOptimizerEngine):
-    """ Performs local optimization of multiobjective function using scipy.
-    The multiobjective function is converted to a scalar by dot product
-    with a weights vector (`weighted_score`).
+class WeightedOptimizerEngine(BaseOptimizerEngine, ScipyOptimizer):
```

Comment:
Can we define `WeightedOptimizerEngine` independently of `ScipyOptimizer`? Otherwise we won't be able to swap the `IOptimizer` mixin for something else.

Reply:
(This links to the comment above)

Reply:
Of course - you can just leave ScipyOptimizer out of the inheritence. As long as you don't try to then make an instance of the class and run it (it will call a method that isn't defined).

Reply:
Or you could replace ScipyOptimizer by some BaseOptimizer, that just has an optimize_function method that returns [1, 1, 1] or something.

Reply:
I don't think this is matter or the right-way or the wrong-way in terms of python. There are dozens of possibilities that are all perfectly valid python. The choice is a matter the wider spec (how these classes are going to be created and used) and taste.

Reply:
> Or you could replace ScipyOptimizer by some BaseOptimizer, that just has an optimize_function method that returns [1, 1, 1] or something.

This would be good, because the engine is only inheriting the minimal methods (optimize_function) and attributes (none!) which are guaranteed to be overwritten by inherited another optimizer.

Reply:
Could we put `optimize_function` as an abstract method on the `WeightedOptimizerEngine`?